siliconcompiler/server.py

```diff
@@ -419,35 +417,40 @@ async def remote_sc_auth(self, chip, username, pk):
             chip.status['decrypt_key'] = base64.urlsafe_b64encode(pk)
             chip.run()
         else:
+            # Reset 'build' directory in NFS storage.
+            _, build_dir = tempfile.mkstemp(prefix=job_hash, suffix=job_nameid)
+
             chip.set('option', 'builddir', build_dir, clobber=True)
             # Run the build command locally.
             from_dir = '%s/%s'%(nfs_mount, job_hash)
-            to_dir   = '/tmp/%s_%s'%(job_hash, job_nameid)
             job_dir  = os.path.join(build_dir, top_module, job_nameid)
             # Write plaintext JSON config to the build directory.
-            subprocess.run(['mkdir', '-p', to_dir])
-            subprocess.run(['mkdir', '-p', '%s/configs'%build_dir])
+            os.makedirs(os.path.join(build_dir, 'configs'), exist_ok=True)
             # Write private key to a file.
             # This should be okay, because we are already trusting the local
             # "compute node" disk to store the decrypted data. Further, the
             # file will be deleted immediately after the run.
             # Even so, use the usual 400 permissions for key files.
-            keypath = f'{to_dir}/pk'
+            keypath = f'{build_dir}/pk'
             with open(os.open(keypath, os.O_CREAT | os.O_WRONLY, 0o400), 'w+') as keyfile:
                 keyfile.write(pk)
             chip.write_manifest(f"{build_dir}/configs/chip{chip.get('option', 'jobname')}.json")
-            # Create the command to run.
-            run_cmd  = f"cp -R {from_dir}/* {to_dir}/ ; "
+
+            utils.copytree(from_dir, build_dir, dirs_exist_ok=True)
+
+            utils.copytree(from_dir, build_dir, dirs_exist_ok=True)
```

Comment:
Do we need both `copytree` calls?

Reply:
looks like I was a little trigger happy with the copy and paste.